txp/Pos/Txp/Settings/Global.hs

```diff
@@ -45,8 +47,22 @@ type TxpGlobalApplyMode ctx m =
 type TxpGlobalRollbackMode m = TxpCommonMode m
 
 -- [CSL-1156] Maybe find better approach (at least wrap into normal types).
-type TxpBlock = Either (Some IsGenesisHeader) (Some IsMainHeader, TxPayload)
-type TxpBlund = (TxpBlock, TxpUndo)
+-- type TxpBlock = Either (Some IsGenesisHeader) (Some IsMainHeader, TxPayload)
+-- type TxpBlund = (TxpBlock, TxpUndo)
+
+-- | Representation of 'Block' passed to a component.
+data ComponentBlock payload =
+    ComponentBlockGenesis (Some IsGenesisHeader)
+    | ComponentBlockMain
+       { bcmHeader  :: !(Some IsMainHeader)
+       , bcmPayload :: !payload }
+
+-- deriving instance Show a => Show (ComponentBlock a)
+
+type TxpBlock = ComponentBlock TxPayload
+type UpdateBlock = ComponentBlock UpdatePayload
+type UndoBlock = ComponentBlock TxpUndo
```

Comment:
No, it doesn't make sense. `Undo` is data which is needed to rollback a block. It's not payload of a block and not part of payload.

`Undo` is whole data necessary to rollback a block, it contains `TxpUndo`, `UpdateUndo` and some other parts. Txp component only needs `TxpUndo`. There is already a type for it, so you can just use it in `TxpBlund` definition (blund = block + undo, just our ad-hoc name).